scipy/integrate/_ivp/ivp.py

```diff
@@ -646,7 +646,7 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     if t_eval is None:
         ts = np.array(ts)
         ys = np.vstack(ys).T
-    else:
+    elif ts:
```

Comment:
Is it worth adding a test based on one of the issue reports? I suppose it might just be a small loop that would have produced an error before but no longer does?

I did notice that in one of the issues there is contention about whether or not this is a bug (I'm not an expert).

Reply:
Well, it is mainly about handling the specific case where the solver exits without having written any solution, a case that both linked issues document. Writing `elif ts:` instead of `else:` avoids calling `numpy.hstack` on an empty list, which otherwise yields an error. Note that this `if` block occurs at the very end of `solve_ivp`, right before the `return` statement. It turns out that `OdeResult`, which is called by the latter `return`, is able to handle empty solutions without any changes.

It is definitely a bug. The contention comes from a disagreement about how `solve_ivp` is used, but it is actually irrelevant to the issue as the error is raised if an event is raised earlier than any `t_eval` value.

I am pushing a commit with a new test.

Reply:
This is indeed a clear bug, and the test looks good to me. The new output seems good:
```
  message: 'A termination event occurred.'
     nfev: 11
     njev: 0
      nlu: 0
      sol: None
   status: 1
  success: True
        t: []
 t_events: [array([7.])]
        y: []
 y_events: [array([[0.41155802, 0.2420958 ]])]
```